packages/commonwealth/client/scripts/views/components/component_kit/cw_cover_image_uploader.tsx

```diff
@@ -0,0 +1,175 @@
+import { MessageRow } from './cw_text_input';
+import { ValidationStatus } from './cw_validation_text';
+
+interface ICWCoverImageUploaderAttrs {
+  headerText?: string;
+  subheaderText?: string;
+  uploadCompleteCallback: CallableFunction;
+}
+
+// TODO Graham 10/24/22: Synchronize avatar upload against new cover upload system
+export default class CWCoverImageUploader
+  implements m.ClassComponent<ICWCoverImageUploaderAttrs>
+{
+  private imageURL: string;
+  private uploading: boolean;
+  private uploadStatus: ValidationStatus;
+
+  async uploadImage(file: File): Promise<[string, ValidationStatus]> {
+    try {
+      const signatureResponse = await $.post(
+        `${app.serverUrl()}/getUploadSignature`,
+        {
+          name: file.name,
+          mimetype: file.type,
+          auth: true,
+          jwt: app.user.jwt,
+        }
+      );
+      if (signatureResponse.status !== 'Success') throw new Error();
+
+      const uploadURL = signatureResponse.result;
+      const uploadResponse = await fetch(uploadURL, {
+        method: 'put',
+        body: file,
+      });
+
+      const imageURL = uploadResponse.url?.replace(/\?.*/, '').trim();
+      if (!imageURL) throw new Error();
+
+      return [imageURL, 'success'];
+    } catch (e) {
+      return [null, 'failure'];
+    }
+  }
+
+  oncreate(vnode: VnodeDOM<ICWCoverImageUploaderAttrs, this>) {
+    const attachZone = document.querySelector('.attach-zone') as HTMLElement;
+    const attachButton = document.querySelector('.attach-btn') as HTMLElement;
+    const pseudoInput = document.querySelector('#pseudo-input') as HTMLElement;
+
+    // Drag'n'Drop helper function
+    const handleDragEvent = (event, hoverAttachZone?: boolean) => {
+      event.preventDefault();
+      event.stopPropagation();
+      attachZone.classList[hoverAttachZone ? 'add' : 'remove']('hovered');
+    };
+
+    const handleUpload = async (file: File) => {
+      const [imageURL, uploadStatus] = await this.uploadImage(file);
+
+      this.uploading = false;
+      this.uploadStatus = uploadStatus;
+
+      if (imageURL) {
+        this.imageURL = imageURL;
+        attachButton.style.display = 'none';
+        vnode.attrs.uploadCompleteCallback(imageURL);
+      }
+
+      m.redraw();
+    };
+
+    // Drag'n'Drop event handler declarations
+    const dragEnterHandler = (enterEvent: DragEvent) => {
+      handleDragEvent(enterEvent, true);
+    };
+
+    const dragOverHandler = (overEvent: DragEvent) => {
+      handleDragEvent(overEvent, true);
+    };
+
+    const dragLeaveHandler = (leaveEvent: DragEvent) => {
+      handleDragEvent(leaveEvent, false);
+    };
+
+    const dropHandler = (dropEvent: DragEvent) => {
+      handleDragEvent(dropEvent, false);
+      delete this.uploadStatus;
+      this.uploading = true;
+      m.redraw();
+      const { files } = dropEvent.dataTransfer;
+      handleUpload(files[0]);
+    };
+
+    attachZone.addEventListener('dragenter', dragEnterHandler);
```

Comment:
The attachZone element would no longer exist at that point, no? I also consulted two code walkthroughs in these designs and didn't see removeEventListener mentioned in either, so gonna assume we're OK unless there's evidence pointing otherwise.

Reply:
we should remove them onremove() lifecycle. just to dot our eyes to ensure no memory leaks

Reply:
This would require me to add 6-10 new state variables to the component. I can do it but it's a heavy-duty dot on our eyes.

Reply:
meh if it bites us we'll do it